content/push/publish.textile

```diff
@@ -0,0 +1,768 @@
+
+h3(#sub-deviceID). Subscibe to channels using @deviceId@
+
+A @deviceId@ is a unique identifier assigned to a device when it first registers with Ably. 
+
+If your application (the client) has the capability to subscribe to push notifications using the @push-subscribe@ "capability":/auth/capabilities, you can directly subscribe to these notifications for a device. This approach allows you to target specific devices with push notifications, ensuring that only the intended recipients receive the messages.
+
+There are two ways to subscribe by @deviceId@: *With* and *without* a @push-subscribe@ capability.
+
+h4. With push-subscribe capabilities
+
+If your application has @push-subscribe@ capability, you can directly subscribe to push notifications for a device:
+
+
+```[android]
+realtime.channels.get("pushenabled:foo").push.subscribeDevice(context);
+
+// or
+
+realtime.channels.get("pushenabled:foo").push.subscribeDeviceAsync(context, new CompletionListener() {
+    @Override
+    public void onSuccess() {}
+
+    @Override
+    public void onError(ErrorInfo errorInfo) {
+        // Handle error.
+    }
+});
+```
+
+```[swift]
+realtime.channels.get("pushenabled:foo").push.subscribeDevice { error
+    // Check error.
+}
+```
+
+```[objc]
+[[realtime.channels get:@"pushenabled:foo"].push subscribeDevice:^(ARTErrorInfo *error) {
+    // Check error.
+}];
+```
+
+h4. Without push-subscribe capabilities
+
+If your application lacks @push-subscribe@ capability, you can register a device for push notifications through the "@activate()@":/api/realtime-sdk/push?langandroid#activate method provided by the @client.push@ object. This method registers the device with the server, and the @deviceId@ will be generated:
+
+```
+client.push.activate();
+```
+
+The "@DeviceDetails@":/api/realtime-sdk/push-admin object contains the @deviceId@, @clientId@, along with other recipient attributes, and it can be accessed once the device is registered.
+
+h3(#sub-clientID). Subscibe to channels using @clientId@
+
+A @clientId@ is used to identify a specific client within a system. This client can be an individual user, a device, or even a group of users or devices that share common permissions or settings.
+
+The @AblyRealtime.push.activate@ method automatically links a @deviceID@ with the @clientId@ from the @AblyRealtime@ instance.
+
+Once a device is associated with a @clientId@, you can subscribe to push notifications for all devices linked to that @clientId@ with a single operation. For instance, if you have multiple devices (like a phone, tablet, and laptop) associated with the same @clientId@, you can efficiently manage their push notification subscriptions together.
+
+On the other hand, if a user no longer uses a device, it is automatically disassociated from the @clientId@, resulting in the loss of its subscription to the channels associated with that @clientId@.
```

Comment:
I think this sentence could be worded a little better, as it could be misconstrued that if a single device is removed, then all other device subscriptions associated with the `clientId` are lost too.